DR-1583 Update resource config to reflect new TDR permissions schema

nmalfroy opened the pull request:
Ticket: https://broadworkbench.atlassian.net/browse/DR-1583
- Followup PR will remove the actions with (deprecated) in their descriptions.
- Admin roles are added that allow alter_policies but these roles are not intended to be granted to end users


---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

nmalfroy committed:
Update resource config to reflect new TDR permissions schema

Note: followup PR will remove the actions with (deprecated) in their descriptions

samanehsan commented on src/main/resources/reference.conf:
```diff
@@ -676,11 +688,26 @@ resourceTypes = {
+      unlink_snapshot = {
+        description = "Delete a snapshot that uses this dataset"
+      }
+      list_snapshots = {
```
Are any of the roles below supposed to have this action?

nmalfroy replied:
Ah yeah, doh.  I missed that.  nice catch!

fboulnois commented on src/main/resources/reference.conf:
```diff
@@ -751,16 +787,19 @@ resourceTypes = {
       }
       reader = {
-        roleActions = ["read_data", "discover_data", "read_policy::custodian"]
+        roleActions = ["read_data", "discover_data", "read_policy::steward", "read_policy::discoverer"]
```
Is this a breaking change?

nmalfroy replied:
I went back and forth on this.  Ultimately, no, since we're getting rid of custodian.  That being said, I'll re-add until I change the UI to not request the custodians

fboulnois reviewed (commented):
One question:

nmalfroy committed:
Address PR feedback

and also re-add snapshot.custodian actions to the role until we move to prod in case there are custodians in dev that expec that access

nmalfroy commented:
jenkins retest

samanehsan reviewed (approved):
LGTM!
